internal/provisioner/objects/deployment/deployment_test.go

```diff
@@ -212,6 +223,44 @@ func TestDesiredDeployment(t *testing.T) {
 	checkDeploymentHasStrategy(t, deploy, cntr.Spec.ContourDeploymentStrategy)
 }
 
+func TestDesiredDeploymentWhenSettingWatchNamespaces(t *testing.T) {
+	testCases := []struct {
+		description    string
+		namespaces     []string
+		expectArgExist bool
+	}{
+		{
+			description:    "several valid namespaces",
+			namespaces:     []string{"ns1", "ns2"},
+			expectArgExist: true,
+		},
+		{
+			description:    "single valid namespace",
+			namespaces:     []string{"ns1"},
+			expectArgExist: true,
+		},
+	}
+
+	for _, tc := range testCases {
+		t.Run(tc.description, func(t *testing.T) {
+			name := "deploy-test"
+			cntr := model.Default(fmt.Sprintf("%s-ns", name), name)
+			icName := "test-ic"
+			cntr.Spec.IngressClassName = &icName
+			// Change the Contour watch namespaces flag
+			cntr.Spec.WatchNamespaces = tc.namespaces
+			deploy := DesiredDeployment(cntr, "ghcr.io/projectcontour/contour:test")
+			container := checkDeploymentHasContainer(t, deploy, contourContainerName, true)
+			arg := fmt.Sprintf("--watch-namespaces=%s", strings.Join(append(tc.namespaces, cntr.Namespace), ","))
+			if tc.expectArgExist {
+				checkContainerHasArg(t, container, arg)
+			} else {
```

Comment:
looks like this case is never actually hit given the tests here, should we make this assertion on the test `TestDesiredDeployment` instead? or add another case with an empty set of namespaces